limits: &limits
    acceleration:
        linear: 5
        angular: 5
    twist:
        linear: 2
        angular: 2

noisy_imu: &noisy_imu
    accelerometer: 0.1 # change the noise on accelerometer
    gyro: 0.1          # change the noise of the gyro
    magnetometer: 0.02

noisy_oswin: &noisy_oswin
    imu_covariances: *noisy_imu
    state:
        pose:
            position:
                x: -5 # initial starting location of the tutle
                y: 0
        twist:
            linear: 0
            angular: 0
    limits: *limits
    publishers:
        imu: true
        pose: true

worlds:
    imu_test:
        turtles:
            oswin: *noisy_oswin

particle_filter: # all of the parameters for the particle filter
    num_particles: 100 # how many particles you have
    resampling_frequency: 1 # How often you resample particles
    use_gps: true # if we should turn on our sensors

    init_x_mean: -5 # the mean of a gaussian to sample the initial x values for particles
    init_x_var: 0.25 # the variance of a gaussian to sample the initial x values for particles
    init_y_mean: 0 # the mean of a gaussian to sample the initial y values for particles
    init_y_var: 0.25 # the variance of a gaussian to sample the initial y values for particles
    init_yaw_mean: 0 # the mean of a gaussian to sample the initial yaw values for particles
    init_yaw_var: 0.1 # the variance of a gaussian to sample the initial yaw values for particles

    imu_sigma_x: 0.05 # variance applied to the motion update in x
    imu_sigma_y: 0.05 # variance applied to the motion update in y
    imu_sigma_yaw: 0.1 # variance applied to the motion update in yaw
    imu_sigma_vx: 0.05 # variance applied to the motion update in velocity in the x direction
    imu_sigma_vy: 0.05 # variance applied to the motion update in velocity in the y direction

    gps_cov: 0.1 # the covariance of the gps measurement

buzzsim_helper:
    gps_mean: 0 # the mean of a gaussian we sample the gps noise from
    gps_variance: 0.1 # variance of a gaussian we sample the noise from
    speed: 0.5 # speed of the turtle
